slave-new/process-o365/bin/process-o365.sh

```diff
@@ -0,0 +1,262 @@
+#!/bin/bash
+
+PROTOCOL_VERSION='3.0.0'
+
+function process {
+	ACCESS_TOKEN=`cat /tmp/o365-access-token`
+
+	LICENSES="licenses.csv"
+	FILE_PHOTOS="photos.csv"
+	FILE_USERS="users.json"
+
+	E_USERS_SYNC=(50 'Error during users synchronization')
+
+	HOME_DIR=`eval echo ~`
+
+	create_lock
+
+	# GET licenses:
+	curl --header "Authorization: $ACCESS_TOKEN" --header "Accept: application/json" 'https://graph.microsoft.com/v1.0/subscribedSkus' | perl -nle 'print "$1,$2" while(/"skuId"\s*:\s*"([^"]*)","skuPartNumber"\s*:\s*"([^"]*)"/g)' >> $HOME_DIR/$LICENSES
+
+	PERL_SYNCHRONIZE_USERS_SCRIPT='
```

Comment:
It's possible (not necessary - only offering the option) to have this script in separate file and execute it from this (process-o365.sh) file.